unison-runtime/src/Unison/Runtime/Machine.hs

```diff
@@ -518,29 +555,9 @@ exec !_ !denv !_activeThreads !stk !k _ (Print i) = do
   t <- peekOffBi stk i
   Tx.putStrLn (Util.Text.toText t)
   pure (denv, stk, k)
-exec !_ !denv !_activeThreads !stk !k _ (Lit (MI n)) = do
-  stk <- bump stk
-  upoke stk n
-  pure (denv, stk, k)
-exec !_ !denv !_activeThreads !stk !k _ (Lit (MD d)) = do
-  stk <- bump stk
-  pokeD stk d
-  pure (denv, stk, k)
-exec !_ !denv !_activeThreads !stk !k _ (Lit (MT t)) = do
-  stk <- bump stk
-  bpoke stk (Foreign (Wrap Rf.textRef t))
-  pure (denv, stk, k)
-exec !_ !denv !_activeThreads !stk !k _ (Lit (MM r)) = do
+exec !_ !denv !_activeThreads !stk !k _ (Lit ml) = do
   stk <- bump stk
-  bpoke stk (Foreign (Wrap Rf.termLinkRef r))
-  pure (denv, stk, k)
-exec !_ !denv !_activeThreads !stk !k _ (Lit (MY r)) = do
-  stk <- bump stk
-  bpoke stk (Foreign (Wrap Rf.typeLinkRef r))
-  pure (denv, stk, k)
-exec !_ !denv !_activeThreads !stk !k _ (BLit rf tt l) = do
-  stk <- bump stk
-  bpoke stk $ buildLit rf tt l
+  poke stk $ litToVal ml
```

Comment:
Is collapsing these branches okay? I presume that since `litToVal` is inlined GHC is probably going to optimize the cases anyways.

Reply:
Yeah, this seems fine.